lib/src/graph.dart

```diff
@@ -270,27 +270,150 @@ class Graph {
   }
 
   /// Finds all subjects which have a certain predicate and object.
-  Set<URIRef> subjects(URIRef pre, dynamic obj) {
+  ///
+  /// If `pre` is provided, it checks for the predicate.
+  /// If `obj` is provided, it checks for the object.
+  /// If both are provided, it checks for both predicate and object.
+  /// If neither is provided, it returns all subjects in the triples.
+  Set<URIRef> subjects({URIRef? pre, dynamic obj}) {
```

Comment:
Is it possible to simplify the code using Dart's *expr1 ?? expr2* [conditional expression](https://dart.dev/language/operators#conditional-expressions)?
For example, does the following cover the four cases you considered?
```
if (t.pre == (pre ?? t.pre) && t.obj == (obj ?? t.obj)) {
    objs.add(t.sub);
}
```
Similar changes might be applicable to the `predicates()` and `objects()` function.